dkg/bcast/impl.go

```diff
@@ -15,22 +15,40 @@ import (
 	"github.com/obolnetwork/charon/p2p"
 )
 
+// Component is the reliable-broadcast handler, in charge of signature and message
+// dispatch.
+type Component struct {
+	BroadcastFunc BroadcastFunc
+
+	allowedMsgIDs map[string]struct{}
+	srv           *server
+	secret        *k1.PrivateKey
+	peers         []peer.ID
+}
+
+// Handle adds a callback for msgID.
+func (c *Component) Handle(msgID string, callback Callback) {
+	c.allowedMsgIDs[msgID] = struct{}{}
+	c.srv.registerCallback(msgID, callback)
+}
+
 // New registers a new reliable-broadcast server and returns a reliable-broadcast client function.
-func New(tcpNode host.Host, peers []peer.ID, secret *k1.PrivateKey,
-	allowedMsgIDs []string, callback Callback,
-) BroadcastFunc {
-	allow := make(map[string]bool)
-	for _, msgID := range allowedMsgIDs {
-		allow[msgID] = true
+func New(tcpNode host.Host, peers []peer.ID, secret *k1.PrivateKey) *Component {
+	c := Component{
+		allowedMsgIDs: map[string]struct{}{},
+		secret:        secret,
+		peers:         peers,
 	}
 
-	signFunc := newK1Signer(secret, allow)
-	verifyFunc := newPeerK1Verifier(peers, allow)
+	signFunc := c.newK1Signer()
+	verifyFunc := c.newPeerK1Verifier()
 
-	_ = newServer(tcpNode, signFunc, verifyFunc, callback)
 	cl := newClient(tcpNode, peers, p2p.SendReceive, p2p.Send, hashAny, signFunc, verifyFunc)
 
-	return cl.Broadcast
+	c.BroadcastFunc = cl.Broadcast
```

Comment:
not sure about having an exposed field called `BroadcastFunc`, rather just add a method called `Broadcast` on the component type and call `cl.Broadcast`